dist/sails.io.js

```diff
@@ -958,6 +964,28 @@ return new(b[["Active"].concat("Object").join("X")])("Microsoft.XMLHTTP")}catch(
 
         self.on('disconnect', function() {
           self.connectionLostTimestamp = (new Date()).getTime();
+          // If there is a list of registered callbacks, call each of them with an error
+          // and then wipe the list.
+          if (typeof self.responseCbs === 'object' && self.responseCbs.length) {
+            self.responseCbs.forEach(function(responseCb) {
```

Comment:
since you are calling a callback you are giving an opportunity for other code to change this array.  I suggest making a copy of the array and iterating on the copy instead unless you are already enforcing immutability elsewhere.  You can replace self.responseCbs immediately before iterating as well so that if, for whatever reason (which should be blocked by the software being offline), if new subscriptions get added they don't get accidentally cleared.
These edge cases are usually handled by a 'safe subscriber' object wrapping the callback instead of directly storing a callback so that you can block the callback directly instead of depending on where it is stored.

Reply:
@pharos42 Good call, thanks for the input.